commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/CommonsLangPortedFractionTest.java

```diff
@@ -0,0 +1,1144 @@
+ *      assertEquals(3, result.getDenominator());
+ *
+ *      f = Fraction.of(-2, -3);
+ *      result = f.reduce();
+ *      assertEquals(2, result.getNumerator());
+ *      assertEquals(3, result.getDenominator());
+ *
+ *      f = Fraction.of(2, -3);
+ *      result = f.reduce();
+ *      assertEquals(-2, result.getNumerator());
+ *      assertEquals(3, result.getDenominator());
+ *
+ *      f = Fraction.of(-2, 3);
+ *      result = f.reduce();
+ *      assertEquals(-2, result.getNumerator());
+ *      assertEquals(3, result.getDenominator());
+ *      assertSame(f, result);
+ *
+ *      f = Fraction.of(2, 3);
+ *      result = f.reduce();
+ *      assertEquals(2, result.getNumerator());
+ *      assertEquals(3, result.getDenominator());
+ *      assertSame(f, result);
+ *
+ *      f = Fraction.of(0, 1);
+ *      result = f.reduce();
+ *      assertEquals(0, result.getNumerator());
+ *      assertEquals(1, result.getDenominator());
+ *      assertSame(f, result);
+ *
+ *      f = Fraction.of(0, 100);
+ *      result = f.reduce();
+ *      assertEquals(0, result.getNumerator());
+ *      assertEquals(1, result.getDenominator());
+ *      assertSame(result, Fraction.ZERO);
+ *
+ *      f = Fraction.of(Integer.MIN_VALUE, 2);
+ *      result = f.reduce();
+ *      assertEquals(Integer.MIN_VALUE / 2, result.getNumerator());
+ *      assertEquals(1, result.getDenominator());
+ *  }
+ *
+ *  @Test
+ *  public void testreciprocal() {
+ *      Fraction f = null;
+ *
+ *      f = Fraction.of(50, 75);
+ *      f = f.reciprocal();
+ *      assertEquals(3, f.getNumerator());
+ *      assertEquals(2, f.getDenominator());
+ *
+ *      f = Fraction.of(4, 3);
+ *      f = f.reciprocal();
+ *      assertEquals(3, f.getNumerator());
+ *      assertEquals(4, f.getDenominator());
+ *
+ *      f = Fraction.of(-15, 47);
+ *      f = f.reciprocal();
+ *      assertEquals(47, f.getNumerator());
+ *      assertEquals(-15, f.getDenominator());
+ *
+ *      assertThrows(ArithmeticException.class, () -> Fraction.of(0, 3).reciprocal());
+ *      Fraction.of(Integer.MIN_VALUE, 1).reciprocal();
+ *
+ *      f = Fraction.of(Integer.MAX_VALUE, 1);
+ *      f = f.reciprocal();
+ *      assertEquals(1, f.getNumerator());
+ *      assertEquals(Integer.MAX_VALUE, f.getDenominator());
+ *  }
+ */
+
+    @Test
+    public void testNegate() {
+        Fraction f = null;
+
+        f = Fraction.of(50, 75);
+        f = f.negate();
+        assertEquals(-2, f.getNumerator());
+        assertEquals(3, f.getDenominator());
+
+        f = Fraction.of(-50, 75);
+        f = f.negate();
+        assertEquals(2, f.getNumerator());
+        assertEquals(3, f.getDenominator());
+
+        // large values
+        f = Fraction.of(Integer.MAX_VALUE - 1, Integer.MAX_VALUE);
+        f = f.negate();
+        assertEquals(Integer.MIN_VALUE + 2, f.getNumerator());
+        assertEquals(Integer.MAX_VALUE, f.getDenominator());
+
+        Fraction.of(Integer.MIN_VALUE, 1).negate();
```

Comment:
Was there an assertion associated with this? Perhaps add some and a comment that this is possible in numbers:
```java
// This is possible in numbers
f = Fraction.of(Integer.MIN_VALUE, 1).negate();
assertEquals(1, f.signum());
assertEquals(Integer.MIN_VALUE, f.getNumerator());
assertEquals(-1, f.getDenominator());
```

Reply:
@aherbert done.